libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs

```diff
@@ -106,7 +106,27 @@ private static (string applicationId, string endpointKey, string endpoint) Parse
                 throw new ArgumentException(nameof(applicationEndpoint));
             }
 
-            var applicationId = uri.Segments.Last();
+            string applicationId = null;
+            var foundApps = false;
+            foreach (var segment in uri.Segments)
```

Comment:
Whenever I'm doing string manipulations by hand I feel bad. I wonder is using the Uri class would be more reasonable, Uri already has a Segments property.

https://docs.microsoft.com/en-us/dotnet/api/system.uri.segments?view=netframework-4.8